src/env_dispatch.cpp

```diff
@@ -195,6 +195,10 @@ void env_dispatch_var_change(const wcstring &key, env_stack_t &vars) {
     s_var_dispatch_table->dispatch(key, vars);
 }
 
+void env_dispatch_var_change_ffi(const wcstring &key) {
+    return env_dispatch_var_change(key, env_stack_t::principal());
```

Comment:
interesting that we don't have a test for this. Fair enough, we'll fix it later

Reply:
there's only one parser at the moment, eventually we'll have more